source/dagger/gameplay/team_game/towers.cpp

```diff
@@ -0,0 +1,88 @@
+#include "towers.h"
+#include "mage.h"
+
+#include "core/engine.h"
+
+using namespace ancient_defenders;
+
+UInt32 ancient_defenders::TowerPlacementInfo::selectedSpot = 0;
+Bool ancient_defenders::TowerPlacementInfo::spotLocked = false;
+Sequence<Vector2> ancient_defenders::TowerPlacementInfo::spotCoordinates = {};
+
+Float32 ancient_defenders::TowerStats::constructionGoal = 10.0f;
+
+void ancient_defenders::TowerBehaviorSystem::SpinUp()
+{
+    Engine::Dispatcher().sink<NextFrame>().connect<&TowerBehaviorSystem::OnEndOfFrame>(this);
+}
+
+void ancient_defenders::TowerBehaviorSystem::WindDown()
+{
+    Engine::Dispatcher().sink<NextFrame>().disconnect<&TowerBehaviorSystem::OnEndOfFrame>(this);
+}
+
+void ancient_defenders::TowerBehaviorSystem::Run()
+{
+    Engine::Registry().view<TowerStats, RangeOfAttack, Animator>().each(
+        [](TowerStats & tower_, RangeOfAttack & range_, Animator & animator_)
+    {
+        if (tower_.constructionProgress < TowerStats::constructionGoal) return;
+        else if (!tower_.constructed) {
+            tower_.constructed = true;
+            TowerPlacementInfo::spotLocked = false;
+
+            auto & sprite = Engine::Registry().emplace<Sprite>(entt::to_entity(Engine::Registry(), tower_));
+            AssignSprite(sprite, "ancient_defenders:" + tower_.type);
+            sprite.position = { TowerPlacementInfo::spotCoordinates[tower_.address].x,TowerPlacementInfo::spotCoordinates[tower_.address].y, 1.0f };
+            sprite.pivot = { 0,0.25f };
+            sprite.scale = { 2,2 };
+        }
+
+        if (range_.targetFound) tower_.currentAction = EAction::Attacking;
+
+        AnimatorPlay(animator_, "ancient_defenders:towers:" + tower_.type);
+        if (tower_.currentAction == EAction::Idling) {}
+        else if (tower_.currentAction == EAction::Attacking) {
+            Engine::Registry().get<Health>(range_.target).currentHealth -= 15.0f*Engine::DeltaTime();
```

Comment:
15.0f should be some const or variable

Reply:
This part of the code will be re written, this is just a placeholder for testing things.